### DISCUSSION TITLE
[Fossology REST API] - Access issue with POST requests

### DISCUSSION THREAD

**Discussion by pratirvce** [fossology/fossology#2095] [General]
I tried to add a folder using REST API and it’s giving me the below error:

Request:
POST https://<*.com>/repo/api/v1/folders?parentFolder=1&folderName=REST&Authorization=<active_token>

Response:
{
    "code": 403,
    "message": "Parent folder can not be accessed!",
    "type": "ERROR"
}

I have been granted admin access, and I am able to get response for all the GET requests, but I am facing problems with POST request. I tried creating new REST API Read/Write access tokens and tried to use that but no luck. Any help regarding this would be highly appreciated. Thank you!

**Comment by GMishx** [comment]
Can you please check if your root folder is actually folder 1? Even as admin, the user can access only those folders which are under the root folder assigned. To check, please goto [Admin > Users > Edit User Account](http://localhost/repo/?mod=user_edit) and check the "Select the user's top-level folder."

![image](https://user-images.githubusercontent.com/18077542/131615998-84e662b6-3d66-408f-9049-126cf9826dd9.png)

It can also be verified in the browse view.

**Reply by pratirvce** [reply]
Thanks for responding to my question. I am able to create the folder from GUI(i.e Organize>Folders>Create) in the root or any subdirectory, so I don't suspect user access issue in the project.

 But REST API create folder is throwing the same below error for every folder (root / or any subdirectory) i try to create a new folder under, so I don't even suspect i m setting the wrong parent folder ID in the api:
{
    "code": 403,
    "message": "Parent folder can not be accessed!",
    "type": "ERROR"
}
Is there a way to debug logs from fossology because I am now suspecting whether the api is returning the right error code/description.

**Reply by dineshr93** [reply]
1.  Check if url is http and https distinction.
2.  Check I'd of Parent folder is 1(click the parent folder and check value of folder Id in the url or make GET request get all folders to check the correct Id
3. This is how I have formed 
https://github.com/dineshr93/easy_fossy/blob/c3023c88b50a6a36a26b54c040e6d97fa0b7d9bb/src/easy_fossy/__init__.py#L395
